apps/ledger-live-mobile/src/actions/settings.ts

```diff
@@ -1,227 +1,454 @@
+);
+export const completeOnboarding = () => completeOnboardingAction();
+
+const installAppFirstTimeAction =
+  createAction<SettingsInstallAppFirstTimePayload>(
+    SettingsActionTypes.SETTINGS_INSTALL_APP_FIRST_TIME,
+  );
+export const installAppFirstTime = (hasInstalledAnyApp: boolean) =>
+  installAppFirstTimeAction({
+    hasInstalledAnyApp,
+  });
+
+const switchCountervalueFirstAction = createAction(
+  SettingsActionTypes.SETTINGS_SWITCH_COUNTERVALUE_FIRST,
+);
+export const switchCountervalueFirst = () => switchCountervalueFirstAction();
+
+const setHideEmptyTokenAccountsAction =
+  createAction<SettingsHideEmptyTokenAccountsPayload>(
+    SettingsActionTypes.SETTINGS_HIDE_EMPTY_TOKEN_ACCOUNTS,
+  );
+export const setHideEmptyTokenAccounts = (hideEmptyTokenAccounts: boolean) =>
+  setHideEmptyTokenAccountsAction({
+    hideEmptyTokenAccounts,
+  });
+
+const blacklistTokenAction = createAction<SettingsBlacklistTokenPayload>(
+  SettingsActionTypes.BLACKLIST_TOKEN,
+);
+export const blacklistToken = (tokenId: string) =>
+  blacklistTokenAction({
+    tokenId,
+  });
+
+const showTokenAction = createAction<SettingsShowTokenPayload>(
+  SettingsActionTypes.SHOW_TOKEN,
+);
+export const showToken = (tokenId: string) =>
+  showTokenAction({
+    tokenId,
+  });
+
+const hideNftCollectionAction = createAction<SettingsHideNftCollectionPayload>(
+  SettingsActionTypes.HIDE_NFT_COLLECTION,
+);
+export const hideNftCollection = (collectionId: string) =>
+  hideNftCollectionAction({
+    collectionId,
+  });
+
+const unhideNftCollectionAction =
+  createAction<SettingsUnhideNftCollectionPayload>(
+    SettingsActionTypes.UNHIDE_NFT_COLLECTION,
+  );
+export const unhideNftCollection = (collectionId: string) =>
+  unhideNftCollectionAction({
+    collectionId,
+  });
+
+const dismissBannerAction = createAction<SettingsDismissBannerPayload>(
+  SettingsActionTypes.SETTINGS_DISMISS_BANNER,
+);
+export const dismissBanner = (bannerId: string) =>
+  dismissBannerAction({
+    bannerId,
+  });
+
+const setCarouselVisibilityAction =
+  createAction<SettingsSetCarouselVisibilityPayload>(
+    SettingsActionTypes.SETTINGS_SET_CAROUSEL_VISIBILITY,
+  );
+export const setCarouselVisibility = (carouselVisibility: {
+  [key: string]: boolean;
+}) => setCarouselVisibilityAction({ carouselVisibility });
+
+const setAvailableUpdateAction =
+  createAction<SettingsSetAvailableUpdatePayload>(
+    SettingsActionTypes.SETTINGS_SET_AVAILABLE_UPDATE,
+  );
+export const setAvailableUpdate = (hasAvailableUpdate: boolean) =>
+  setAvailableUpdateAction({
+    hasAvailableUpdate,
+  });
+
+const setThemeAction = createAction<SettingsSetThemePayload>(
+  SettingsActionTypes.SETTINGS_SET_THEME,
+);
+export const setTheme = (theme: Theme) =>
+  setThemeAction({
+    theme,
+  });
+
+const setOsThemeAction = createAction<SettingsSetOsThemePayload>(
+  SettingsActionTypes.SETTINGS_SET_OS_THEME,
+);
+export const setOsTheme = (osTheme: string) =>
```

Comment:
There are a lot of places where string could be replaced by enums - and it's not the purpose of this PR to rewrite this kind of things.
It is a good point though and it would be good to have later on - on a follow up PR.